File: API/LuminaAPI/Model/PMS/ProductDetail.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace LuminaAPI.Model.PMS
{
    public class ProductDetail
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string _id { get;set; }

        // Public properties
        public string ProductID { get; set; }

        public string? ProductName { get; set; }

        public bool IsActive { get; set; }

        public int BrandID { get; set; }

        public int ModelID { get; set; }

        [BsonIgnore]
        public string Brand {  get; set; }

        [BsonIgnore]
        public string Model {  get; set; }  
    }

}

File: UI/LuminaUI/src/app/shared/Models/ProductDetail.ts
export class ProductDetail {
    public _id: string;
    public ProductID: string;
    public ProductName: string;
    public ModelID: string;
    public IsActive: boolean;
  
    constructor(
      _id: string,
      ProductID: string,
      ProductName: string,
      ModelID: string,
      IsActive: boolean,
    ) {
      this._id = _id;
      this.ProductID = ProductID;
      this.ProductName = ProductName;
      this.ModelID = ModelID;
      this.IsActive = IsActive;
    }
  }
  